src/NServiceBus.Persistence.AzureTable/Outbox/OutboxRecord.cs

```diff
@@ -44,5 +46,13 @@ public void SetAsDispatched()
             Operations = Array.Empty<TransportOperation>();
             DispatchedAt = DateTimeOffsetHelper.ToWireFormattedString(DateTimeOffset.UtcNow);
         }
+
+        class StorageTransportOperation
```

Comment:
Does moving from `TransportOperation` to `StorageTransportOperation` impact wire compatibility? e.g. let's imagine that an endpoint is upgraded to V8 and at startup look for not yet dispatched messages. Would that work?

Reply:
Yep, it should. StorageTransportOperation is basically the V7 representation of those classes. After the change from a Dictionary<string, string> to DispatchProperties, JSON.NET would serialize DispatchProperties correctly as a dictionary, but would not deserialize the "dictionary" back to DispatchProperties. JSON.NET does not do any of the `$type` properties unless you have a situation like List<BaseClass> that is full of instances of more specialized classes that inherit from BaseClass. So on the wire, it's all the same.